spacy/errors.py

```diff
@@ -417,6 +417,10 @@ class Errors(object):
             "component matches the model being loaded.")
     E150 = ("The language of the `nlp` object and the `vocab` should be the same, "
             "but found '{nlp}' and '{vocab}' respectively.")
+    E151 = ("Trying to call nlp.update without required annotation types. "
+            "Expected top-level keys 'words', 'tags', 'heads', 'deps', 'entities' or 'cats'."
+            " Got: {unexpected_keys}.")
+
```

Comment:
Could we somehow make sure to define this list only once? Because also here we'd want to include "links" (cf other comment)

Reply:
Yes, that's a good idea! We can just pass in the `expected_keys` here.

Reply:
Good ideas! I put both of these in the latest commit

Reply:
Looks good!